homeassistant/components/growatt_server/config_flow.py

```diff
@@ -24,7 +24,11 @@ def __init__(self):
     def _async_show_user_form(self, errors=None):
         """Show the form to the user."""
         data_schema = vol.Schema(
-            {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+            {
+                vol.Required(CONF_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+                vol.Optional(CONF_URL, default=DEFAULT_URL): str,
```

Comment:
Do we know which servers exist and provide a choice instead?

Reply:
Unfortunately not. The best solution I could come up with (other than having to submit a PR everytime someone comes up with a new one) was to provide a default (which is used for all new Growatt systems) and then allow people to override it.

Reply:
I think a list is easier to use for end-users. A free form field requires technical knowledge of knowing which endpoint your system is using.

So, I would vote for the list and extend it over time when questions are asked

Reply:
I've pushed changes in alignment with your suggestion and feedback from the Growatt users in the community forums.
Just hoping the tests now pass (I have manually checked everything at my end already)